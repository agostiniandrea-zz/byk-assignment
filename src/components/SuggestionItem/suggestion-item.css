.suggestion-item {
  position: relative;
  width: 100%;
  height: 30px;
  border: 1px solid grey;
  box-sizing: border-box;
}

.value {
  position: absolute;
  font-size: 0.8em;
  left: 5px;
  transform: translateY(-50%);
}