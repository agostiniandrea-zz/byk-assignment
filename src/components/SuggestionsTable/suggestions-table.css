.suggestions-table {
  position: relative;
  width: 100%;
  height: auto;
}