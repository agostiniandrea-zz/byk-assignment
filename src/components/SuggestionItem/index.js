import React from 'react';
import './suggestion-item.css';

class SuggestionItem extends React.Component {
  getStrongString(input) {
    return `<strong>${input}</strong>`;
  }

  renderedString() {
    let newString = "";
    if (this.props.searchterm && this.props.nrResults) {
      if (this.props.typedString && this.props.searchterm.includes(this.props.typedString)) {
        const searchTerm = this.props.searchterm;
        const typedString = this.props.typedString;
        const stringSlices = searchTerm.split(typedString);
        if (searchTerm.startsWith(typedString)) {
          newString = typedString + this.getStrongString(stringSlices[1]);
        } else if (searchTerm.endsWith(typedString)) {
          newString = this.getStrongString(stringSlices[0]) + typedString;
        } else {
          newString = this.getStrongString(stringSlices[0]) + typedString + this.getStrongString(stringSlices[1]);;
        }
      } else {
        newString = `<strong>${this.props.searchterm}</strong>`;
      }
      newString += ` (${this.props.nrResults})`
    }
    return newString;
  }

  render() {
    return (
      <div className="suggestion-item">
        <p className="value" dangerouslySetInnerHTML={{ __html: this.renderedString() }} />
      </div>
    );
  }
};

export default SuggestionItem;