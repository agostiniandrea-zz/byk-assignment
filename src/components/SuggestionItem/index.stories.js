import React from 'react';
import SuggestionItem from './index';

export default { title: 'SuggestionItem' };

const mockData = require("../../../api/_search.get.json");
const suggestions = mockData.suggestions;
const mySuggestion = suggestions[0];

export const withoutText = () => <SuggestionItem />;
export const withText = () => <SuggestionItem nrResults={mySuggestion.nrResults} searchterm={mySuggestion.searchterm} typedString={"trui"} />;