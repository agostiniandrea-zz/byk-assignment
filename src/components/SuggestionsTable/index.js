import React from 'react';
import './suggestions-table.css';
import SuggestionItem from '../SuggestionItem';

class SuggestionsTable extends React.Component {
  render() {
    return (
      <div className="suggestions-table">
        {this.props.items && this.props.items.map((object, i) =>
          <SuggestionItem nrResults={object.nrResults} searchterm={object.searchterm} typedString={this.props.typedString} key={i} />
        )}
      </div>
    );
  }
};

export default SuggestionsTable;