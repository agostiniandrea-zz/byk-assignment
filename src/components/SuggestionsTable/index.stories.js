import React from 'react';
import SuggestionsTable from './index';

export default { title: 'SuggestionsTable' };

const mockData = require("../../../api/_search.get.json");
const suggestions = mockData.suggestions;

export const withoutItems = () => <SuggestionsTable />;
export const withItems = () => <SuggestionsTable items={suggestions} typedString="trui" />;