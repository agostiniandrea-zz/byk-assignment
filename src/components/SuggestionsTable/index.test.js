import React from 'react';
import { shallow } from 'enzyme';
import SuggestionsTable from './index';
import SuggestionItem from '../SuggestionItem';

const mockData = require("../../../api/_search.get.json");
const suggestions = mockData.suggestions;

let wrapper1;
let wrapper2;

beforeEach(() => {
  wrapper1 = shallow(<SuggestionsTable />);
  wrapper2 = shallow(<SuggestionsTable items={suggestions} typedString="trui" />);
})

afterEach(() => {
  wrapper1.unmount();
  wrapper2.unmount();
});

describe("component/SuggestionsTable/index.js", () => {
  it('renders without crashing (no props)', () => {
    expect(wrapper1).toMatchSnapshot();
    expect(wrapper1.find(SuggestionItem).length).toBe(0);
  });

  it('renders without crashing (with props)', () => {
    expect(wrapper2).toMatchSnapshot();
    expect(wrapper2.find(SuggestionItem).length).toBe(suggestions.length);
  });
});