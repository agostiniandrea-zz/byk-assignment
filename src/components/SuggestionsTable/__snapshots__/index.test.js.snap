// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component/SuggestionsTable/index.js renders without crashing (no props) 1`] = `ShallowWrapper {}`;

exports[`component/SuggestionsTable/index.js renders without crashing (with props) 1`] = `ShallowWrapper {}`;
